engine/gamesys/src/gamesys/components/comp_tilegrid.cpp

```diff
@@ -631,6 +632,39 @@ namespace dmGameSystem
         return where;
     }
 
+    static void RenderListFrustumCulling(dmRender::RenderListVisibilityParams const &params)
+    {
+        DM_PROFILE("TileGridFrustrumCulling");
+        TileGridWorld* tilegrid_world = (TileGridWorld*)params.m_UserData;
+        const dmIntersection::Frustum frustum = *params.m_Frustum;
+        uint32_t num_entries = params.m_NumEntries;
+        for (uint32_t i = 0; i < num_entries; ++i)
+        {
+            dmRender::RenderListEntry* entry = &params.m_Entries[i];
+            uint32_t index, layer, region_x, region_y;
+            //entry->m_UserData - encoded region info
+            DecodeGridAndLayer(entry->m_UserData, index, layer, region_x, region_y);
+            TileGridComponent* component = tilegrid_world->m_Components[index];
+
+            TileGridResource* resource = component->m_Resource;
+            TextureSetResource* texture_set = GetTextureSet(component);
+            int32_t tile_width = (int32_t)texture_set->m_TextureSet->m_TileWidth;
+            int32_t tile_height = (int32_t)texture_set->m_TextureSet->m_TileHeight;
+
+            int32_t column_count = (int32_t)resource->m_ColumnCount;
+            int32_t row_count = (int32_t)resource->m_RowCount;
+            int32_t min_x = resource->m_MinCellX + region_x * TILEGRID_REGION_SIZE;
+            int32_t min_y = resource->m_MinCellY + region_y * TILEGRID_REGION_SIZE;
+            int32_t max_x = dmMath::Min(min_x + (int32_t)TILEGRID_REGION_SIZE, resource->m_MinCellX + column_count);
```

Comment:
Will `column_count` ever be < 1?
And even if it is, we can set it to 1?
I feel that the two cases inside Min() are a bit tricky to read. One uses "min_x", and the other uses "m_MinCellX".

Reply:
`column_count` can not be < 1. I cast all unsigned type to signed because at last step here https://github.com/defold/defold/pull/8935/files/fa8d2d61102b20660bcee6144fba1c7e79b6ef4d#diff-840abe2ff7226d24a7cc3b2884131009a9cc28f9a062011b163f31a83c1608ebR661-R662 I ran into issue that signed `min_...` that < 0 cast to unsigned and I got very huge number. That why I converted all variables to signed types.